app/src/example_app.cpp

```diff
@@ -0,0 +1,148 @@
+        "%zu @%" PRIi64 "\n",
+        var->topic().data(),
+        client,
+        var->len(),
+        now);
+
+    if (var->topic() == "/flecs/flunder/cpp/int") {
+        const auto i = std::atoll(var->value().data());
+        std::fprintf(stdout, "\tValue: %lld\n", i);
+    } else if (var->topic() == "/flecs/flunder/cpp/double") {
+        const auto d = std::atof(var->value().data());
+        std::fprintf(stdout, "\tValue: %lf\n", d);
+    } else if (var->topic() == "/flecs/flunder/cpp/string") {
+        std::fprintf(stdout, "\tValue: %s\n", var->value().data());
+    } else if (var->topic() == "/flecs/flunder/cpp/timestamp") {
+        const auto t1 = std::stoll(var->value().data());
+        const auto diff = now - t1;
+        std::fprintf(stdout, "\tMessage sent @%lld (%lld ns ago)\n", t1, diff);
+    }
+}
+
+auto flunder_receive_callback_userp(
+    flunder::client_t* client,
+    const flunder::variable_t* var,
+    const void* userp) //
+    -> void
+{
+    const auto timestamp = std::chrono::high_resolution_clock::now().time_since_epoch().count();
+    std::fprintf(
+        stdout,
+        "Received flunder message for topic %s on client %p with length "
+        "%zu and userdata %s "
+        "@%" PRIi64 "\n",
+        var->topic().data(),
+        client,
+        var->len(),
+        static_cast<const char*>(userp),
+        timestamp);
+}
+
+auto exec_example_loop() //
+    -> int
+{
+    auto flunder_client = flunder::client_t{};
+
+    do {
+        if (flunder_client.connect() == -1) {
+            std::fprintf(stdout, "Connection could not be established, waiting 5s for reconnect.\n");
+            std::this_thread::sleep_for(std::chrono::seconds(5));
+            continue;
+        }
+        std::fprintf(stdout, "Connection was established.\n");
+
+        flunder_client.add_mem_storage("flunder-cpp", "/flecs/flunder/**");
+
+        flunder_client.subscribe("/flecs/flunder/cpp/**", &flunder_receive_callback);
+        const char* userdata = "Hello, world!";
+        flunder_client.subscribe(
+            "/flecs/flunder/external",
+            &flunder_receive_callback_userp,
+            static_cast<const void*>(userdata));
+
+        while (!g_stop && flunder_client.is_connected()) {
+            constexpr auto i = 1234;
+            flunder_client.publish("/flecs/flunder/cpp/int", i);
+
+            constexpr auto d = 3.14159;
+            flunder_client.publish("/flecs/flunder/cpp/double", d);
+
+            constexpr auto str = "Hello, world!";
+            flunder_client.publish("/flecs/flunder/cpp/string", str);
+
+            const auto t = std::chrono::high_resolution_clock::now().time_since_epoch().count();
+            flunder_client.publish("/flecs/flunder/cpp/timestamp", t);
+
+            std::this_thread::sleep_for(std::chrono::seconds(5));
+        };
+
+        flunder_client.unsubscribe("**");
+        flunder_client.disconnect();
+
+        std::this_thread::sleep_for(std::chrono::seconds(5));
```

Comment:
Remove sleep. Blocks shutdown of App after SIGTERM / Ctrl-C for 5 seconds.

Reply:
"Remove unnecessary sleep in loop"

![image](https://github.com/FLECS-Technologies/apps-tech.flecs.ample/assets/86266568/404bae6d-3d51-49cf-9975-6fb2c83f0624)